### DISCUSSION TITLE
New utility to help migrate to V3

### DISCUSSION THREAD

**Discussion by ryangoree** [muhammadsammy/tailwindcss-classnames#352] [Show and tell]
I made a utility package which converts your codebase from single `classname`/`tw` functions to [utility functions](https://github.com/muhammadsammy/tailwindcss-classnames/issues/293) :)

It was pretty rushed, but worked well for our codebase. If it helps you or if you have any feedback, I'd love to hear about it!

[https://www.npmjs.com/package/twcn3](https://www.npmjs.com/package/twcn3)

**README:**

# twcn3

A cli utility for converting codebases upgrading from [tailwindcss-classnames](https://www.npmjs.com/package/tailwindcss-classnames) v2 to v3.

In tailwindcss-classnames v3, you're required to use [new utility functions](https://github.com/muhammadsammy/tailwindcss-classnames/issues/293) to compose your classes rather than a single `classname` or `tw` function. Depending on the size of your codebase, this can be an incredibly time-consuming task.

This utility does it with one command:

```
npx twcn3 convert
```

## Options

- **`--version`** - Show version number
- **`--help`** - Show help
- **`-t`, `--types <path-to-types>`** - The path to your built tailwindcss-classnames types file. Only required if you have custom classes.
- **`-d`, `--dir <path-to-src>`** - The path to the directory with the files to be converted.
(**default: `src`**)
- **`-a`, `--alias <function-name>`** - The name used when importing the classnames function from your type file. (i.e. <code>import <strong>tw</strong> from 'tailwindcss-classnames.ts'</code>) (**default: `tw`**)

**Comment by muhammadsammy** [comment]
Nice work!
I will include this in the README. Thank you!
